### PR TITLE
Fixing incorrect H2 URL

### PR DESCRIPTION
**Pull request by gunnarmorling**

### CODE REVIEW

File: docs/src/main/asciidoc/datasource-guide.adoc [hunk 1]
```diff
@@ -100,7 +100,7 @@ A viable configuration file would be:
 
 [source,properties]
 --
-quarkus.datasource.url=jdbc:h2:tcp://localhost/mem:default
+quarkus.datasource.url=jdbc:h2:mem:default
```
Comment (gsmet) [inline docs/src/main/asciidoc/datasource-guide.adoc]: I don't think that's incorrect. We use a similar one in our integration tests.

Did you have any issue with it?

(Note: we don't support really support embedded H2 as it doesn't work in native mode)

**Comment by gunnarmorling** [comment]
Let's say it failed due H2 not being available on localhost. Seems like an
unnecessary stumbling stone in this guide.

>

**Comment by gsmet** [comment]
Well, it's the one that will work in native and I don't think we want to advertise something that won't work at all in native mode.

**Comment by gunnarmorling** [comment]
Ok, feel free to close it then. One could think about splitting it up, too. For me it was just a short WTH when going through this article, esp. as it doesn't talk about launching the server before.

**Comment by gsmet** [comment]
This guide is about configuring the datasources, not about launching the server.

Closing, thanks anyway.

**Pull request closed without merging**